### ISSUE TITLE
Caddyfile: Add "resolvers" as a subdirective of "tls"

### ISSUE THREAD

**Issue by francislavoie** [caddyserver/caddy#4008]
Currently, `resolvers` only exists as an option of `issuer` https://caddyserver.com/docs/caddyfile/directives/tls#issuers. If a user wants to configure `resolvers`, then they need to do it like this, to manually propagate the `resolvers` to all the relevant issuers:

```
tls {
	issuer zerossl {
		resolvers 8.8.8.8:53
		dns gandi {env.GANDI_APIKEY}
	}
	issuer acme {
		resolvers 8.8.8.8:53
		dns gandi {env.GANDI_APIKEY}
	}
}
```

Most `tls` options like `email` and `dns` get copied down to the default issuers, if no issuers are specified. A user on Discord wanted the same to be done for `resolvers` as well.

Basically, the ideal config would be this, to avoid repetition and specifying the issuers:

```
tls {
	dns gandi {env.GANDI_APIKEY}
	resolvers 8.8.8.8:53
}
```

**Comment by mholt** [comment]
Easy breezy.

**Comment by MagnuM2K** [comment]
Thanks!
This works like a charm.
Anyway to specify the order of issuer, though?

I had put ZeroSSL as primary (just because I could.. ;) ) and now with this change, it grabs a cert from LE.

**Comment by mholt** [comment]
@MagnuM2K It honors the order you specify in your config. To get more help, please ask on our forum: https://caddy.community - and fill out the template. Thanks!
